subprojects/tooling-api/src/main/java/org/gradle/tooling/internal/consumer/connection/ParameterValidatingConsumerConnection.java

```diff
@@ -65,4 +66,12 @@ private void validateParameters(ConsumerOperationParameters operationParameters)
             }
         }
     }
+
+    private void validateBuildActionParameters(ConsumerOperationParameters operationParameters) {
+        if (!targetVersionDetails.supportsRunTasksBeforeExecutingAction()) {
+            if (operationParameters.getTasks() != null) {
+                throw Exceptions.unsupportedFeature("run tasks before executing BuildAction feature", targetVersionDetails.getVersion(), "3.5");
```

Comment:
This is an error message shown to a developer, so we should point them at the name of the method they called. I'd replace "run tasks before executing BuildAction feature" with "BuildActionExecuter.forTasks()"